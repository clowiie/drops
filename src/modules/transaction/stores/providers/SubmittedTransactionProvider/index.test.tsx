import { render } from '@testing-library/react'

import * as SubmittedTransactionProviderType from '.'

describe('SubmittedTransactionProvider', () => {
  const useSubmittedTransactionSpy = jest.fn()
  jest.doMock(
    'modules/transaction/hooks/useSubmittedTransaction',
    () => useSubmittedTransactionSpy,
  )

  const Component = () => <h1>Component</h1>

  const { default: SubmittedTransactionProvider } =
    require('.') as typeof SubmittedTransactionProviderType

  beforeEach(() => {
    useSubmittedTransactionSpy.mockReturnValue({
      submittedHash: 'hash',
      setSubmittedHash: () => {},
      clearSubmitted: () => {},
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <SubmittedTransactionProvider>
          <Component />
        </SubmittedTransactionProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Component')).toBeDefined()
    })
  })
})
