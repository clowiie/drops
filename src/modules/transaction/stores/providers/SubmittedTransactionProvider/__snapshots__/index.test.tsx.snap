// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmittedTransactionProvider snapshots should match snapshots 1`] = `
<h1>
  Component
</h1>
`;
